test/hooks.ts

```diff
@@ -340,8 +341,7 @@ test('beforeRetry hook is never called for the initial request', async t => {
 				hooks: {
 					beforeRetry: [
 						({options}) => {
-							// @ts-expect-error @TODO
-							options.headers!.set('unicorn', fixture);
+							(options.headers as Partial<Headers> | undefined)?.set?.('unicorn', fixture);
```

Comment:
Why is this needed? Is the `options.headers` type not correct?

Reply:
You need to narrow the type to work with it

`options.headers` here can be either of union type element: `Headers | string[][] | Record<string, string>;`

But `set` function exists only in `Headers`, that's why I cast it here

Reply:
I think `NormalizedOptions` needs to ensure headers are only `Headers`. We should present simple unified types to the user.